clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -871,6 +871,34 @@ float3 degrees(float3);
 _HLSL_BUILTIN_ALIAS(__builtin_hlsl_elementwise_degrees)
 float4 degrees(float4);
 
+//===----------------------------------------------------------------------===//
+// distance builtins
+//===----------------------------------------------------------------------===//
+
+/// \fn K distance(T X, T Y)
+/// \brief Returns a distance scalar between two vectors of \a X and \a Y.
```

Comment:
the distance builtin seems to work both on vector and scalar inputs. Seems like this comment would better belong at line 891.

Reply:
This comment is for the whole set of  distance overloads. We have never commented per function to this point.

Reply:
The scalar overload conflicts with the comment. Wouldn't
"Returns a distance scalar between two vectors or scalars X and Y" be better?

Reply:
I see what you are saying. The comment was copied from the official documentation and technically the scalar version is just `abs(X-Y)`. In past intrinsics I've just used this first line: https://github.com/MicrosoftDocs/win32/blob/docs/desktop-src/direct3dhlsl/dx-graphics-hlsl-distance.md?plain=1#L20

I could instead use this line for return value as it makes no mention of vectors: https://github.com/MicrosoftDocs/win32/blob/docs/desktop-src/direct3dhlsl/dx-graphics-hlsl-distance.md?plain=1#L46

Reply:
Yes, either of those 2 options would be good!